crates/iota-indexer/src/config.rs

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) Mysten Labs, Inc.
+// Modifications Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{net::SocketAddr, path::PathBuf};
+
+use clap::{Args, Parser, Subcommand};
+use iota_names::config::IotaNamesConfig;
+use iota_types::base_types::{IotaAddress, ObjectID};
+use url::Url;
+
+use crate::db::ConnectionPoolConfig;
+
+#[derive(Parser, Clone, Debug)]
+#[clap(
+    name = "IOTA indexer",
+    about = "An off-fullnode service serving data from IOTA protocol"
+)]
+pub struct IndexerConfig {
+    #[clap(long, alias = "db-url")]
```

Comment:
The `clap` attribute seems deprecated. Please use `arg` instead and other attributes according to the latest [derive reference](https://docs.rs/clap/latest/clap/_derive/index.html)